### PR TITLE
Fix synthetic type bounds in Tasty reflect

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/tastyreflect/TastyImpl.scala [hunk 1]
```diff
@@ -538,7 +538,7 @@ object TastyImpl extends scala.tasty.Tasty {
 
     object Synthetic extends SyntheticExtractor {
       def unapply(x: TypeTree)(implicit ctx: Context): Boolean = x match {
-        case Trees.TypeTree() => true
+        case x @ Trees.TypeTree() => !x.tpe.isInstanceOf[Types.TypeBounds]
         case _ => false
       }
     }
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/tastyreflect/TastyImpl.scala]: Given the name `synthetic`, I cannot justify the change. Maybe a doc for `Synthetic`, I naively assume synthetic bounds are also synthetic.

File: library/src/scala/tasty/util/ShowExtractors.scala [hunk 2]
```diff
@@ -188,7 +190,8 @@ class ShowExtractors[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
       case Type.PolyType(argNames, argBounds, resType) =>
         this += "Type.PolyType(" ++= argNames += ", " ++= argBounds += ", " += resType += ")"
       case Type.TypeLambda(argNames, argBounds, resType) =>
-        this += "Type.TypeLambda(" ++= argNames += ", " ++= argBounds += ", " += resType += ")"
+        // resType is not printed to avoid cycles
```
Comment (liufengyun) [inline library/src/scala/tasty/util/ShowExtractors.scala]: I'm curious what kind of cycles can happen here?

Comment (nicolasstucki) [reply library/src/scala/tasty/util/ShowExtractors.scala]: For definitions, we use the tree of the definition. Somewhere in the body of the type lambda there was a reference to itself and then tried to reprint itself in a loop.

File: compiler/src/dotty/tools/dotc/tastyreflect/TastyImpl.scala (hunk 1 above)
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/tastyreflect/TastyImpl.scala]: I will add some documentation.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/tastyreflect/TastyImpl.scala]: Added

**Pull request merged**